rmgpy/data/kinetics/familyTest.py

```diff
@@ -180,6 +182,289 @@ def testReactBenzeneBond2(self):
         self.assertEqual(len(products), 1)
         self.assertTrue(expectedProduct.isIsomorphic(products[0]))
 
+    def testIntraHMIgration(self):
+        """
+        Test that the intra_H_migration family, which is its own reverse, returns a properly re-labelled product structure.
+        """
+        family = self.database.families['intra_H_migration']
+        reactants = [Molecule().fromAdjacencyList("""
+multiplicity 2
+1  *2 C u0 p0 c0 {3,S} {11,S} {12,S} {13,S}
+2  *6 C u0 p0 c0 {4,S} {5,S} {6,D}
+3  *7 C u0 p0 c0 {1,S} {7,D} {14,S}
+4  *1 C u1 p0 c0 {2,S} {8,S} {15,S}
+5     C u0 p0 c0 {2,S} {10,D} {17,S}
+6  *4 C u0 p0 c0 {2,D} {7,S} {19,S}
+7  *5 C u0 p0 c0 {3,D} {6,S} {21,S}
+8     C u0 p0 c0 {4,S} {9,D} {16,S}
+9     C u0 p0 c0 {8,D} {10,S} {20,S}
+10    C u0 p0 c0 {5,D} {9,S} {18,S}
+11 *3 H u0 p0 c0 {1,S}
+12    H u0 p0 c0 {1,S}
+13    H u0 p0 c0 {1,S}
+14    H u0 p0 c0 {3,S}
+15    H u0 p0 c0 {4,S}
+16    H u0 p0 c0 {8,S}
+17    H u0 p0 c0 {5,S}
+18    H u0 p0 c0 {10,S}
+19    H u0 p0 c0 {6,S}
+20    H u0 p0 c0 {9,S}
+21    H u0 p0 c0 {7,S}
+""")]
+        expectedProduct = Molecule().fromAdjacencyList("""
+multiplicity 2
+1  *1 C u1 p0 c0 {3,S} {12,S} {13,S}
+2  *6 C u0 p0 c0 {4,S} {5,S} {6,D}
+3  *7 C u0 p0 c0 {1,S} {7,D} {14,S}
+4  *2 C u0 p0 c0 {2,S} {11,S} {8,S} {15,S}
+5     C u0 p0 c0 {2,S} {10,D} {17,S}
+6  *4 C u0 p0 c0 {2,D} {7,S} {19,S}
+7  *5 C u0 p0 c0 {3,D} {6,S} {21,S}
+8     C u0 p0 c0 {4,S} {9,D} {16,S}
+9     C u0 p0 c0 {8,D} {10,S} {20,S}
+10    C u0 p0 c0 {5,D} {9,S} {18,S}
+11 *3 H u0 p0 c0 {4,S}
+12    H u0 p0 c0 {1,S}
+13    H u0 p0 c0 {1,S}
+14    H u0 p0 c0 {3,S}
+15    H u0 p0 c0 {4,S}
+16    H u0 p0 c0 {8,S}
+17    H u0 p0 c0 {5,S}
+18    H u0 p0 c0 {10,S}
+19    H u0 p0 c0 {6,S}
+20    H u0 p0 c0 {9,S}
+21    H u0 p0 c0 {7,S}
+""")
+        products = family.applyRecipe(reactants)
+
+        self.assertEqual(len(products), 1)
+        self.assertTrue(expectedProduct.isIsomorphic(products[0]))
```

Comment:
These tests pass despite the labels having been wrong because `isIsomorphic` does not check atom labels. In order to check those, an mapping needs to be passed as a second argument to `isIsomorphic`.